spec.md

```diff
@@ -2714,6 +2718,75 @@ need only respect the 2-state behavior of a circuit.  This is a limitation on
 the scope of what behavior is observable (i.e., a relaxation of the
 ["as-if"](https://en.wikipedia.org/wiki/As-if_rule) rule).
 
+## Indeterminate Values
+
+An indeterminate value represents a value which is unknown or unspecified.  
+Indeterminate values are generally implementation defined, with constraints 
+specified below.  An indeterminate value may be assumed to be any specific 
+value (not necessarily literal), at an implementation's discretion, if, in doing
+so, all observable behavior is as if the indeterminate value always took the 
+specific value.
+
+This allows transformations such as the following, where when `a` has an 
+indeterminate value, the implementation chooses to consistently give it a value 
+of 'v'.  An alternate, legal mapping, lets the implementaiton give it the value
+`42`.  In both cases, there is no visibility of `a` when it has an indeterminate
+value which is not mapped to the value the implementaiton choose.
+
+``` firrtl
+module IValue :
+  output o : UInt<8>
+  input c : UInt<1>
+  input v : UInt<8>
+
+  wire a : UInt<8>
+  a is invalid
+  when c :
+    a <= v
+  o <= a
+```
+is transformed to:
+``` firrtl
+module IValue :
+  output o : UInt<8>
+  input c : UInt<1>
+
+  o <= v
+```
+Note that it is equally correct to produce:
+``` firrtl
+module IValue :
+  output o : UInt<8>
+  input c : UInt<1>
+
+  wire a : UInt<8>
+  when c :
+    a <= v
+   else :
+     a <= UInt<3>("h42")
+  o <= a
+```
+
+The behavior of constructs which cause indeterminate values is implementation 
+defined with the following constraints.  
+* Register initialization is done in a consistent way for all registers.  If 
```

Comment:
Mega-nit: the rest of the doc uses `-`, not `*` for itemized lists, and indents the rest of the paragraph to match the start of the item.